### PR TITLE
feat: add typescript example

### PR DESCRIPTION
**Pull request by franzmoro**
This is a leftover from the [original PR](https://github.com/isaachinman/next-i18next/pull/922) addressing missing TS types. It adds a typescript example.

Used default `tsconfig.json` created when running `yarn dev` straight after having created the Next.js app with `npx create-next-app`. I could be up to the dev(s) to modify this config by themselves.

### CODE REVIEW

**Comment by isaachinman** [comment]
My main concern with adding other examples has always been maintenance. However, I think TS is so mainstream now, that I would actually consider just making the main example TS, as the source code itself isn't too different.

What are your thoughts?

**Comment by franzmoro** [comment]
The code of the 2 examples is virtually the same, yes.
I agree also on TS being mainstream, but still a lot of teams don't use it. I'm not sure if having just the TS example becomes confusing, as JS is supported of course.

I'm happy to keep both, but I'm not too opinionated about it - just trying to help. LMK.

File: examples/with-typescript/components/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { version } from 'next-i18next/package.json'
+import { useTranslation } from 'next-i18next'
+
+export const Footer: React.FC<{}> = () => {
```
Comment (isaachinman) [inline examples/with-typescript/components/Footer.tsx:5]: The generic argument is unnecessary – you can just write `React.FC =`.

File: examples/with-typescript/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import Head from 'next/head'
+
+interface Props {
```
Comment (isaachinman) [inline examples/with-typescript/components/Header.tsx:4]: Let's use `type` over `interface ` in all cases.

File: examples/with-typescript/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "v10",
```
Comment (isaachinman) [inline examples/with-typescript/package.json:2]: Name should be something like `next-i18next/examples/with-typescript`.

File: examples/with-typescript/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+
+import { useTranslation } from 'next-i18next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+import { Header } from '../components/Header'
+import { Footer } from '../components/Footer'
+
+const Homepage = () => {
```
Comment (isaachinman) [inline examples/with-typescript/pages/index.tsx:10]: Needs `React.FC`.

File: examples/with-typescript/pages/second-page.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Link from 'next/link'
+import { useTranslation } from 'next-i18next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+import { Header } from '../components/Header'
+import { Footer } from '../components/Footer'
+
+const SecondPage: React.FC<{}> = () => {
```
Comment (isaachinman) [inline examples/with-typescript/pages/second-page.tsx:9]: Again, unnecessary generic arg can be dropped.

**Review by isaachinman — CHANGES_REQUESTED** [review]

File: examples/with-typescript/pages/_app.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { appWithTranslation } from 'next-i18next'
```
Comment (sethreidnz) [inline examples/with-typescript/pages/_app.tsx:2]: Why is this called appWithTranslation? When I try to import this in my TS project like this I get:

```
'"../../node_modules/next-i18next/types"' has no exported member named 'appWithTranslation'. Did you mean 'AppWithTranslation'?
```

Comment (sethreidnz) [reply examples/with-typescript/pages/_app.tsx:2]: It looks like the original example has a file `i18n.js` that it imports this function from. But in this example that file doesn't exist... and this function is being imported from `next-i18next` how does that work?

Comment (sethreidnz) [reply examples/with-typescript/pages/_app.tsx:2]: Ahh I get it now, this is part of a new beta version.

File: examples/with-typescript/components/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  title: string;
+}
+
+export const Header: React.FC<Props> = ({ title }) => (
```
Comment (danielrakoczyas) [inline examples/with-typescript/components/Header.tsx:8]: HeaderProps

**Pull request closed without merging**